src/integrationtests/java/com/aws/greengrass/integrationtests/RateLimiterTest.java

```diff
@@ -103,13 +106,15 @@ void GIVEN_throttled_cloud_update_requests_WHEN_cloud_updates_THEN_cloud_updates
                         .lastSyncedDocument(lastSyncedDocument.getBytes())
                         .cloudVersion(0).build()));
         lenient().when(dao.updateSyncInformation(any(SyncInformation.class))).thenReturn(true);
+        RequestBlockingQueue queue = spy(kernel.getContext().get(RequestBlockingQueue.class));
+        kernel.getContext().put(RequestBlockingQueue.class, queue);
 
         startNucleusWithConfig(NucleusLaunchUtilsConfig.builder().configFile("rateLimits.yaml").mockCloud(true).mockDao(true).build());
         SyncHandler syncHandler = kernel.getContext().get(SyncHandler.class);
         JsonNode updateDocument = JsonUtil.getPayloadJson(localShadowContentV1.getBytes()).get();
 
         assertThat("syncing has started", () -> kernel.getContext().get(RealTimeSyncStrategy.class).isSyncing(), eventuallyEval(is(true)));
-
+        verify(queue, timeout(2000).times(1)).clear();
```

Comment:
As part of the test, 10 requests are added and processed from the queue. But before all the requests are processed from the queue, some of these are cleared from the when the sync starts up. 

https://github.com/aws-greengrass/aws-greengrass-shadow-manager/blob/33173fa05614ce22deca4f36ebcd363845688095/src/main/java/com/aws/greengrass/shadowmanager/sync/SyncHandler.java#L158


So this change waits for the queue to clear up when it is synced for the first time before adding and processing the requests from the queue.